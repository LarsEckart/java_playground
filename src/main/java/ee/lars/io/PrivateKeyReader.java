package ee.lars.io;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.security.KeyFactory;
import java.security.NoSuchAlgorithmException;
import java.security.PrivateKey;
import java.security.spec.EncodedKeySpec;
import java.security.spec.InvalidKeySpecException;
import java.security.spec.PKCS8EncodedKeySpec;

public class PrivateKeyReader {

    /**
     * Reads a private key file which is in PKCS#8 format.
     * For example create a private key via
     * openssl genrsa -out private_key.pem 2048
     * and then convert it via
     * openssl pkcs8 -topk8 -inform PEM -outform DER -in private_key.pem -out private_key.der -nocrypt
     */
    public static PrivateKey get(String filename)
            throws IOException, NoSuchAlgorithmException, InvalidKeySpecException
    {
        byte[] keyBytes = Files.readAllBytes(Paths.get(filename));

        EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(keyBytes);
        KeyFactory keyFactory = KeyFactory.getInstance("RSA");
        final PrivateKey privateKey = keyFactory.generatePrivate(keySpec);
        return privateKey;
    }
}
