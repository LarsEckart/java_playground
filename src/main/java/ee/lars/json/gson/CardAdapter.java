package ee.lars.json.gson;

import com.google.gson.TypeAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonToken;
import com.google.gson.stream.JsonWriter;
import ee.lars.json.Card;
import ee.lars.json.Suit;
import java.io.IOException;

public class CardAdapter extends TypeAdapter<Card> {

    @Override
    public void write(JsonWriter writer, Card card) throws IOException {
        if (card == null) {
            writer.nullValue();
            return;
        }
        String json = card.rank + card.suit.name().substring(0, 1);
        writer.value(json);
    }

    @Override
    public Card read(JsonReader reader) throws IOException {
        if (reader.peek() == JsonToken.NULL) {
            reader.nextNull();
            return null;
        }
        String card = reader.nextString();

        if (card.length() != 2) {
            throw new IllegalArgumentException("Unknown card: " + card);
        }

        char rank = card.charAt(0);
        switch (card.charAt(1)) {
            case 'C':
                return new Card(rank, Suit.CLUBS);
            case 'D':
                return new Card(rank, Suit.DIAMONDS);
            case 'H':
                return new Card(rank, Suit.HEARTS);
            case 'S':
                return new Card(rank, Suit.SPADES);
            default:
                throw new IllegalArgumentException("unknown suit: " + card);
        }
    }
}