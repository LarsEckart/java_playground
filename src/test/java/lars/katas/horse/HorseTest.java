package lars.katas.horse;

import com.google.gson.Gson;
import org.approvaltests.Approvals;
import org.junit.Test;

import java.io.IOException;
import java.util.Collections;
import java.util.List;
import java.util.Optional;

//@RunWith(FasterTestCommitRevertRunner.class)
public class HorseTest {

    @Test
    public void FilterSortPaginate_No_Filters_No_Sorting_No_Pagination() throws IOException {
        // Arrange - this is data from another service or database
        List<String> headers = SampleHorseData.GetSampleHeaders();
        List<List<Object>> tableData = SampleHorseData.GetSampleTableData();

        // These objects describe the query we got from the front end
        List<FilterMetadata> filters = Collections.emptyList();
        Optional<SortMetadata> sortMetadata = Optional.empty();
        PaginationMetadata paginationMetadata = new PaginationMetadata(0, 9);

        // Act
        PaginatedTable table = Horse.FilterSortPaginateTable(headers, tableData, filters, sortMetadata, paginationMetadata);

        Gson gson = new Gson();
        String json = gson.toJson(table);
        // Assert the data to be sent to the front end
        Approvals.verifyJson(json);
    }
}