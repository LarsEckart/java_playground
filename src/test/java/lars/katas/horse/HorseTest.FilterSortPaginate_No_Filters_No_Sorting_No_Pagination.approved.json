{
  "headers": [
    "Breed",
    "Colour",
    "Height",
    "Age",
    "Shoes"
  ],
  "tableData": [
    [
      "Thoroughbred",
      "Bay",
      "1.6",
      "3",
      "true"
    ],
    [
      "Thoroughbred",
      "Grey",
      "1.55",
      "3",
      "true"
    ],
    [
      "Arabian horse",
      "Bay",
      "1.51",
      "5",
      "true"
    ],
    [
      "Shetland Pony",
      "Black",
      "1.01",
      "2",
      "false"
    ],
    [
      "Shire horse",
      "Black",
      "1.71",
      "4",
      "true"
    ]
  ],
  "totalRows": 5
}